parquet/src/main/java/org/apache/iceberg/parquet/ParquetTypeVisitor.java

```diff
@@ -34,9 +34,13 @@
   protected Deque<String> fieldNames = Lists.newLinkedList();
 
   public static <T> T visit(Type type, ParquetTypeVisitor<T> visitor) {
+    return visit(type, visitor, false);
+  }
+
+  public static <T> T visit(Type type, ParquetTypeVisitor<T> visitor, Boolean ignoreRepeated) {
```

Comment:
I add an extra parameter here because the field names built from the original `ParquetTypeVisitor` are different in name mapping. For example, if we have a field consists of a list of the map type like below:
```
optional(1, "list_of_maps",Types.ListType.ofOptional(2, Types.MapType.ofOptional(3, 4,
    Types.StringType.get(),Types.StringType.get());
```
the field names of the map ke and value are `list_of_map.list.element.map.key` and `list_of_map.list.element.map.value`. While the names in name mapping are `list_of_map.element.key` and `list_of_map.element.value`.

Since the `NameMapping` is used by Avro already, so I update `ParquetTypeVisitor` a bit to handle this.

Reply:
Thanks for the explanation, this will be useful when I take a closer look.

Reply:
I think we should use an alternative approach. I opened #1001 for this.

The problem with this is that it adds a parameter to customize behavior, when that should be the responsibility of the visitor. This is also a bit confusing because it isn't clear why some callers would want to skip repeated fields. It is better to delegate this to the visitor.

The PR I opened allows you to do that. It adds before and after callbacks for the repeated fields (element and key_value), as well as for the element, key, and value types. By default, those call `beforeField` and `afterField`, which are used to maintain the field names list. With that change, you can simply override the repeated element and repeated key value calls to do nothing:

```java

  @Override
  public void beforeRepeatedElement(Type element) {
    // do not add the repeated element's name
  }

  @Override
  public void afterRepeatedElement(Type element) {
    // do not remove the repeated element's name
  }

  @Override
  public void beforeRepeatedKeyValue(Type keyValue) {
    // do not add the repeated element's name
  }

  @Override
  public void afterRepeatedKeyValue(Type keyValue) {
    // do not remove the repeated element's name
  }
```

That passes all of the tests.

Reply:
Great! I will update after the PR merged.